File: Day-08/ContainerWithMostWater.py
class ContainerWithMostWater(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        n= len(height)
        i = 0
        j= n-1

        #al = 0
        al =[]
        while i<j:
            area = min(height[i],height[j]) * (j-i)
            #al=max(al,area)
            al.append(area)
        
            if height[i]<height[j]:
                i+= 1
            else:
                j-=1

      
        return max(al)
        
            

File: Day-29/ContainerWithMostWater.java
class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int max=0;
        int currentArea=0;
        int l=0,r=height.length-1;
        while(l<r){
            if(height[l]<=height[r]){
                currentArea=height[l]*(r-l);
                l++;
            }else{
                currentArea=height[r]*(r-l);
                r--;
            }
            if(max<currentArea){
                max=currentArea;
            }
        }
        return max;
    }
}